### PR TITLE
uint multiplication for flip, flap n' flop

### PR DESCRIPTION
**Pull request by MrChico**

### CODE REVIEW

**Comment by xwvvvvwx** [comment]
Why?

File: src/heal.sol [hunk 1]
```diff
@@ -71,8 +71,10 @@ contract Vow is DSNote {
         z = x - y;
         require(z <= x);
     }
-    function mul(uint x, uint y) internal pure returns (uint z) {
-        require(y == 0 || (z = x * y) / y == x);
+    function mul(uint x, uint y) internal pure returns (int z) {
```
Comment (asymmetric) [inline src/heal.sol]: wut? why does this one get changed to the supposedly uglier implementation?

File: src/heal.sol [hunk 2]
```diff
@@ -116,14 +118,12 @@ contract Vow is DSNote {
     // Debt settlement
     function heal(uint wad) public note {
         require(wad <= Joy() && wad <= Woe());
-        require(int(mul(wad, ONE)) >= 0);
-        VatLike(vat).heal(bytes32(bytes20(address(this))), bytes32(bytes20(address(this))), int(mul(wad, ONE)));
+        VatLike(vat).heal(bytes32(bytes20(address(this))), bytes32(bytes20(address(this))), mul(wad, ONE));
```
Comment (asymmetric) [inline src/heal.sol]: Oh so it's about not having the type conversion?

Comment (MrChico) [reply src/heal.sol]: Or rather, factoring out the type conversion to the multiplication function. I'm fine with omitting the second commit though.

Comment (asymmetric) [reply src/heal.sol]: I slightly prefer the consistent implementation option FWIW.

**Comment by asymmetric** [comment]
Closed because already in https://github.com/makerdao/dss/commit/0e67ddccd73818278f7674880961f2ef9d08e065?

**Comment by MrChico** [comment]
Yea

**Pull request closed without merging**